client/src/components/ProfileEdit.jsx

```diff
@@ -7,94 +7,94 @@ import { Link } from 'react-router-dom'
-
-        if (this.state.new_password === this.state.repeat_password) {
-            const data = {
-                "first_name": this.state.f_name,
-                "last_name": this.state.l_name,
-                "nickname": this.state.nickname,
-                "location": this.state.location,
-                "password": this.state.new_password,
-                "id": this.state.user_edit.id
+        if (this.state.new_password_input ===
+            this.state.repeat_password_input) {
+            if (this.state.nickname_input === "") {
+                this.setState({
+                    errors_edit: {
+                        ...this.state.errors_edit,
+                        nickname: "You can't save empty nickname"
+                    }
+                });
             }
-            request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
-            .then(data => {
-                try {
-                    if ('errors' in data) {
-                        for (let i = 0; i < data['errors'].length; i++) {
-                            if (data.errors[i].name === "first_name") {
-                                this.setState({
-                                    error_first: data.errors[i].description
-                                })
-                            }
-                            if (data.errors[i].name === "last_name") {
-                                this.setState({
-                                    error_last: data.errors[i].description
-                                })
-                            }
-                            if (data.errors[i].name === "nickname") {
-                                this.setState({
-                                    error_nick: data.errors[i].description
-                                })
-                            }
-                        };
+            else {
+                const data = {
+                    "first_name": this.state.first_name_input,
+                    "last_name": this.state.last_name_input,
+                    "nickname": this.state.nickname_input,
+                    "location": this.state.location_input,
+                    "password": this.state.new_password_input,
+                    "id": this.state.user.id
+                }
+                request('/profile/' + this.props.match.params.profile_id,
+                        "PUT", JSON.stringify(data))
+                .then(data => {
+                    if(!data.success) {
+                        if ('errors' in data) {
+                            let errors = data['errors'].reduce(
+                                (accumulator, item) =>
+                                { accumulator[item.name] =
+                                  item.description;
+                                  return accumulator; }, {});
+                            this.setState({ errors_edit: errors });
+                        }
+                        else {
```

Comment:
Please do not ignore my comments. I already asked, how it's possible that `data.success` is false, but no errors?